src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4261,6 +4892,226 @@ .maxstack  1
 ");
         }
 
+        [Fact]
+        public void SupportedBinaryOperators()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static (bool, bool, bool, bool, bool, bool) DoCompare(delegate*<void> func_1a, delegate*<string, void> func_1b)
+    {
+        return (func_1a == func_1b,
+                func_1a != func_1b,
+                func_1a > func_1b,
+                func_1a >= func_1b,
+                func_1a < func_1b,
+                func_1a <= func_1b);
+    }
+
+    static void M(delegate*<void> func_1a, delegate*<string, void> func_1b, delegate*<int> func_2, int* int_1, int* int_2)
+    {
+        var compareResults = DoCompare(func_1a, func_1b);
+        Console.WriteLine(""func_1a == func_1b: "" + compareResults.Item1);
```

Comment:
>compareResults.Item1 [](start = 53, length = 20)

Just curious: Why are the operators for the first pair evaluated with `DoCompare()` and the remaining pairs evaluated inline?

Reply:
I didn't think it was necessary or useful to verify the exact IL for each type of binary operator conversion, so I split out the operators for just the function pointer to function pointer operators and verified this one function. I left a comment for that a bit lower, but I probably should have put it here.

---
In reply to: [413345822](https://github.com/dotnet/roslyn/pull/43355#discussion_r413345822) [](ancestors = 413345822)

Reply:
Actually, I think it was here, but subsequent test additions moved it.

---
In reply to: [413346707](https://github.com/dotnet/roslyn/pull/43355#discussion_r413346707) [](ancestors = 413346707,413345822)